app/routes/_auth+/reset-password.tsx

```diff
@@ -64,16 +64,20 @@ export async function loader({ request }: DataFunctionArgs) {
 
 export async function action({ request }: DataFunctionArgs) {
 	const formData = await request.formData()
-	const result = ResetPasswordSchema.safeParse(
-		preprocessFormData(formData, ResetPasswordSchema),
-	)
-	if (!result.success) {
-		return json({ status: 'error', errors: result.error.flatten() } as const, {
-			status: 400,
-		})
+	const submission = parse(formData, {
+		schema: ResetPasswordSchema,
+		acceptMultipleErrors: () => true,
+	})
+	if (!submission.value) {
+		return json({
+			status: 'error',
+			submission,
+		} as const)
 	}
-
-	const { password } = result.data
+	if (submission.intent !== 'submit') {
+		return json({ status: 'success', submission } as const)
+	}
```

Comment:
We can remove if statement and update the previous one with:
```js
	if (!submission.value || submission.intent !== 'submit') {
```

Reply:
First of all...Thanks for the feedback. Didn't try the async validation for schema before.

Good point. Done